fail2ban/server/utils.py

```diff
@@ -0,0 +1,199 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: t -*-
+# vi: set ft=python sts=4 ts=4 sw=4 noet :
+
+# This file is part of Fail2Ban.
+#
+# Fail2Ban is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# Fail2Ban is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Fail2Ban; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+__author__ = "Serg G. Brester (sebres) and Fail2Ban Contributors"
+__copyright__ = "Copyright (c) 2004 Cyril Jaquier, 2011-2012 Yaroslav Halchenko, 2012-2015 Serg G. Brester"
+__license__ = "GPL"
+
+import logging, os, fcntl, subprocess, time, signal
+from ..helpers import getLogger
+
+# Gets the instance of the logger.
+logSys = getLogger(__name__)
+
+# Some hints on common abnormal exit codes
+_RETCODE_HINTS = {
+	127: '"Command not found".  Make sure that all commands in %(realCmd)r '
+			'are in the PATH of fail2ban-server process '
+			'(grep -a PATH= /proc/`pidof -x fail2ban-server`/environ). '
+			'You may want to start '
+			'"fail2ban-server -f" separately, initiate it with '
+			'"fail2ban-client reload" in another shell session and observe if '
+			'additional informative error messages appear in the terminals.'
+	}
+
+# Dictionary to lookup signal name from number
+signame = dict((num, name)
+	for name, num in signal.__dict__.iteritems() if name.startswith("SIG"))
+
+class Utils():
```

Comment:
>  > +class Utils():
>      without making code even more confusing
>    I've not understood...

in our obscure camelCased notation, leading capital is only for
ClassNames.  Methods, functions, modules should not have a leading
capital letter.

>    two advantages:
>    1) all static methods are moved but we will correct imports only (not a
>    large code change requires)

you are already introducing big changes ;)

>    2) if tomorrow python get a module named utils, it will not confused with
>    him, because of Utils

I would not worry about that ATM (even if introduced, relative imports
would save us)